expertise/models/multifacet_recommender/specter.py

```diff
@@ -304,14 +329,34 @@ def load_emb_file(emb_file):
             print(len(bad_id_set))
             return emb_tensor, id_list, bad_id_set
 
-        with open(publications_path) as f_in:
-            print('Loading cached publications...')
-            paper_emb_train, train_id_list, train_bad_id_set = load_emb_file(f_in)
-            paper_num_train = len(train_id_list)
+        def load_from_redis():
+            paper_emb_size_default = 768
+            id_list = self.pub_note_id_to_title.keys()
+            emb_list = []
+            bad_id_set = set()
+            con = redisai.Client(host='localhost', port=6379, db=5)
+            for paper_id in id_list:
+                try:
+                    paper_emb = con.tensorget(key=paper_id, as_numpy_mutable=True)
+                    assert len(paper_emb) == paper_emb_size_default
+                    emb_list.append(paper_emb)
+                except Exception as e:
+                    bad_id_set.add(paper_id)
+
+            emb_tensor = torch.tensor(emb_list, device=torch.device('cpu'))
```

Comment:
why do you do this? it doesn't seem to be part of the previous code. Do you need this for caching?

Reply:
It's there in the previous code as well. Check [`load_emb_file`](https://github.com/openreview/openreview-expertise/blob/a2a552a6e028e23fab81866bde29328fc46171db/expertise/models/multifacet_recommender/specter.py#L310)

Reply:
yes, I see that, but where do you decide how to load the embeddings? calling to either `load_from_redis` or `load_emb_file`?

Reply:
[Here](https://github.com/openreview/openreview-expertise/blob/a2a552a6e028e23fab81866bde29328fc46171db/expertise/models/multifacet_recommender/specter.py#L354) when loading publications, `load_from_redis` is called, but when loading submissions, `load_emb_file` is called.

Reply:
I would prefer to have the caching logic in one single place. Is it possible to check this when the embedding is being computed?:

```
if cache.exists(publication.id):
     embeddings = cache.get(publication.id)
else:
     embeddings = compute_embeddings(publication.id)
     cache.set(publication.id, embeddings)
```

Maybe this is difficult to add giving the current implementation.